### PR TITLE
Issue 4469 (#16)

### PR DESCRIPTION
**Pull request by kchapple**
This adds Audit Log logging to the patient-merge process.  Keeps track of which user did the merge, which patients were merged and the actions performed delete/update/etc on which table. (It does not log the raw SQL).

Note: I keep seeing a fail in a specific test, but not due to any of the code ... it fails looking for phpunit (which I see in the filesystem) it is linked to another spot, and I assume that is the problem- the link is not being created, or not being followed.  I've re-run the tests a couple of times, and it seems consistent.

@StuartSmith183 

* #4469 Patient Merge Audit Log

Added direct event audit lines for each merge event.

* #4469

Minor cleanup of comments etc.

* #4469 Missed log spots.

Missed some spots where logging should occur. Added those spots in.
Added $target_pid to deleteRows for passing that down to the log method.

* #4469 Moved logMergeEvent

Didn't like it being mashed in between the other methods,
moved it down to the bottom.

* #4469

Added use line for EventAuditLogger.

* #4469

Changed patient-merge to patient_merge
patient-merge gets a bit mangled when querying the log
for the 'event-type' ... the '-' is used to trim the last part off.

* #4469

Messing with the log messages, to make sure they read logically.

* #4469

Added additional tagging.

* #4469

Massive code formatting, for lint checking.

* #4469 : Formatting.

Ah, finally found the right format specification (ci/phpcs.xml)

Co-authored-by: Stuart <geektoast@gmail.com>

Fixes #

### CODE REVIEW

File: interface/patient_file/merge_patients.php [hunk 1]
```diff
@@ -4,24 +4,28 @@
...
  * @author    Rod Roark <rod@sunsetsystems.com>
  * @author    Brady Miller <brady.g.miller@gmail.com>
- * @copyright Copyright (c) 2013 Rod Roark <rod@sunsetsystems.com>
- * @copyright Copyright (c) 2018 Brady Miller <brady.g.miller@gmail.com>
- * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ * @copyright 2013 Rod Roark <rod@sunsetsystems.com>
+ * @copyright 2018 Brady Miller <brady.g.miller@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE
+ * GNU General Public License 3
+ * @link      http://www.open-emr.org
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:15]: lets keep the current header tags intact/standardized unless community decides to change them up

File: interface/patient_file/merge_patients.php [hunk 2]
```diff
@@ -313,27 +483,49 @@ function mergeRows($tblname, $colname, $source_pid, $target_pid)
    <input type='hidden' name='form_source_pid' value='0' />
   </td>
   <td>
-    <?php echo xlt('This is the chart that is to be merged into the main chart and then deleted.'); ?>
+    <?php echo xlt(
+        'This is the chart that is to be merged into the main chart 
+        and then deleted.'
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:488]: ensure don't break up translation constant into multiple line. need to make these one line no matter how long they are (or else will break in the translation pipeline)

File: interface/patient_file/merge_patients.php [hunk 3]
```diff
@@ -313,27 +483,49 @@ function mergeRows($tblname, $colname, $source_pid, $target_pid)
...
-<p><input type='submit' class="btn btn-primary" name='form_submit' value='<?php echo xla('Merge'); ?>' /></p>
+<p><input type='submit' class="btn btn-primary" name='form_submit' 
+    value='<?php echo xla('Merge'); ?>' /></p>
 </div>
 </form>
 <div class="jumbotron">
-    <p class="font-weight-bold"><?php echo xlt('This utility is experimental. Back up your database and documents before using it!'); ?></p>
+    <p class="font-weight-bold"><?php echo xlt(
+        'This utility is experimental. 
+    Back up your database and documents before using it!'
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:500]: ensure don't break up translation constant into multiple lines. need to make these one line no matter how long they are (or else will break in the translation pipeline)

File: interface/patient_file/merge_patients.php [hunk 4]
```diff
@@ -313,27 +483,49 @@ function mergeRows($tblname, $colname, $source_pid, $target_pid)
...
+
+<p><?php echo xlt(
+    'The second ("source") chart will have its demographics, 
+    history and insurance sections discarded.  
+    Its other data will be merged into the target chart.'
+); ?></p>
+
+<p><?php echo xlt(
+    'The merge will not run unless SSN and DOB for the two charts are identical.
+     DOBs cannot be empty.'
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:527]: For this line and 3 more above
ensure don't break up translation constant into multiple lines. need to make these one line no matter how long they are (or else will break in the translation pipeline)

**Comment by StuartSmith183** [comment]
Thanks for the feedback, I'll take care of these. 
In the future ... since these were getting flagged as error/warning for phpcs (and I didn't realize the NLS needed them to be exact), is there something we could/should do in these instances ... something that will let phpcs and/or devs know that the line breaks the standard, but is OK? (// phpcs:ignore or similar, maybe?)

$ phpcs --standard=ci/phpcs.xml  ./interface/patient_file/merge_patients.php

FILE: /home/stuart/Documents/Projects/Medical Information Integration LLC/repos/openemr/interface/patient_file/merge_patients.php
--------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 7 WARNINGS AFFECTING 7 LINES
--------------------------------------------------------------------------------------------------------------------------------------------------------------
   1 | WARNING | A file should declare new symbols (classes, functions, constants, etc.) and cause no other side effects, or it should execute logic with
     |         | side effects, but should not do both. The first symbol is defined on line 80 and the first side effect is on line 16.
 461 | WARNING | Line exceeds 120 characters; contains 248 characters
 488 | WARNING | Line exceeds 120 characters; contains 140 characters
 494 | WARNING | Line exceeds 120 characters; contains 214 characters
 496 | WARNING | Line exceeds 120 characters; contains 193 characters
 498 | WARNING | Line exceeds 120 characters; contains 180 characters
 500 | WARNING | Line exceeds 120 characters; contains 126 characters
--------------------------------------------------------------------------------------------------------------------------------------------------------------

( The header was 100% my goof -- I'd pointed the 'standard' in my IDE to the wrong folder, so it fell back to using the phpcs default standard at first... I fixed all of the code to clear the errors/warning and then realized my configuration mistake, I rolled it back but I guess I needed to go one more revision. :facepalm: )

**Comment by sunsetsystems** [comment]
#4502 also changes this script, but shouldn't be a big deal to fix any conflicts.

**Comment by kchapple** [comment]
bump

File: interface/patient_file/merge_patients.php (hunk 2 above)
Comment (bradymiller) [reply interface/patient_file/merge_patients.php:520]: still needs to be addressed

File: interface/patient_file/merge_patients.php [hunk 5]
```diff
@@ -4,8 +4,8 @@
  * This script merges two patient charts into a single patient chart.
  * It is to correct the error of creating a duplicate patient.
  *
+ * @category  Patient_Data
  * @package   OpenEMR
- * @link      http://www.open-emr.org
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:8]: this line should be added back

**Comment by bradymiller** [comment]
just noted 2 more minor issues to fix:
- adding link back to header
- one more constant that needs to placed all in 1 line

**Comment by kchapple** [comment]
Thanks @bradymiller Fixed.

**Comment by bradymiller** [comment]
hi @kchapple , looks good and bringing in. thanks! -brady

**Pull request merged**